lib/pos_blockchain.go

```diff
@@ -1807,6 +1807,13 @@ func (bc *Blockchain) GetUncommittedTipView() (*UtxoView, error) {
 // getUtxoViewAtBlockHash builds a UtxoView to the block provided. It does this by
 // identifying all uncommitted ancestors of this block and then connecting those blocks.
 func (bc *Blockchain) getUtxoViewAtBlockHash(blockHash BlockHash) (*UtxoView, error) {
+	if viewAtHash, exists := bc.blockViewCache.Lookup(blockHash); exists {
```

Comment:
Can you add a safety check here by block height? If the provided blockHash is at or below the committed tip height then we shouldn't return the cached view. That way, the behavior for returning cached view is identical to line 1826 below where we generate a new view too.

My fear is that we unintentionally call `getUtxoViewAtBlockHash` for a block hash that's below the committed tip and get a cached view back that's stale

Reply:
Great call. I'll make that change and remove the item from the cache if it's there.